dynet/tensor.cc

```diff
@@ -94,11 +95,13 @@ float TensorTools::access_element(const Tensor& v, int index) {
 }
 
 float TensorTools::access_element(const Tensor& v, const Dim& index) {
+  if (v.device->type == DeviceType::CPU) {
+    return (*v)(index[0], index[1]);
+  } else {
 #if HAVE_CUDA
```

Comment:
Similarly here, I think we should check if the device is GPU, and throw an error if we get an unknown device.